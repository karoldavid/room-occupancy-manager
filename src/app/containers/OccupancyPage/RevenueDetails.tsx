import React from 'react';
import { Revenue } from './types';
import { Card } from 'antd';

export interface RevenueDetailsProps {
  data: Revenue;
  title: string;
}

export function RevenueDetails({ data, title }: RevenueDetailsProps) {
  return (
    <Card title={title} bordered={false}>
      <p>{`free: ${data.free}`}</p>
      <p>{`usage: ${data.usage}`}</p>
      <p>{`revenue: ${data.total} ${data.currency}`}</p>
    </Card>
  );
}
