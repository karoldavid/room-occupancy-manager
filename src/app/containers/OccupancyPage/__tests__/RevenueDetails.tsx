import React from 'react';
import { render } from '@testing-library/react';
import { RevenueDetails } from '../RevenueDetails';

const renderRevenueDetails = (
  props: Parameters<typeof RevenueDetails>[number],
) => render(<RevenueDetails {...props} />);

describe('<RevenueDetails />', () => {
  it('should match the snapshot', () => {
    const revenueDetails = renderRevenueDetails({
      data: { free: 4, usage: 3, total: 1234, currency: 'EURO' },
      title: 'Test',
    });
    expect(revenueDetails.container.firstChild).toMatchSnapshot();
  });

  it('should have the correct number of <p> elements', () => {
    const revenueDetails = renderRevenueDetails({
      data: { free: 4, usage: 3, total: 1234, currency: 'EURO' },
      title: 'Test',
    });
    expect(revenueDetails.container.querySelectorAll('p').length).toBe(3);
  });

  it('should have props displayed', () => {
    const free = 7;
    const usage = 4;
    const total = 2345;
    const currency = 'EURO';
    const title = 'Test';

    const revenueDetails = renderRevenueDetails({
      data: { free, usage, total, currency },
      title,
    });

    expect(revenueDetails.queryByText(title)).toBeInTheDocument();
  });
});
