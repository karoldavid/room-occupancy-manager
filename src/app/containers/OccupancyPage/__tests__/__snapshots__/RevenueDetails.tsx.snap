// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RevenueDetails /> should match the snapshot 1`] = `
<div
  class="ant-card"
>
  <div
    class="ant-card-head"
  >
    <div
      class="ant-card-head-wrapper"
    >
      <div
        class="ant-card-head-title"
      >
        Test
      </div>
    </div>
  </div>
  <div
    class="ant-card-body"
  >
    <p>
      free: 4
    </p>
    <p>
      usage: 3
    </p>
    <p>
      revenue: 1234 EURO
    </p>
  </div>
</div>
`;
